protocol-definitions/Jet.yaml

```diff
@@ -321,3 +321,98 @@ methods:
           nullable: false
           since: 2.6
           doc: ''
+  - id: 17
+    name: uploadJobMetaData
+    since: 2.6
+    doc: ''
+    request:
+      retryable: true
+      partitionIdentifier: -1
+      params:
+        - name: sessionId
+          type: UUID
+          nullable: false
+          since: 2.6
+          doc: 'Unique session ID of the job upload request'
+        - name: fileName
+          type: String
+          nullable: false
+          since: 2.6
+          doc: 'Name of the jar file without extension'
+        - name: sha256Hex
+          type: String
+          nullable: false
+          since: 2.6
+          doc: 'Hexadecimal SHA256 of the jar file'
+        - name: snapshotName
+          type: String
+          nullable: true
+          since: 2.6
+          doc: 'Name of the initial snapshot to start the job from'
+        - name: jobName
+          type: String
+          nullable: true
+          since: 2.6
+          doc: 'Name of the job'
+        - name: mainClass
+          type: String
+          nullable: true
+          since: 2.6
+          doc: 'Fully qualified name of the main class inside the JAR file'
+        - name: jobParameters
+          type: List_String
+          nullable: false
+          since: 2.6
+          doc: 'Arguments to pass to the supplied jar file'
+    response:
+      params:
+        - name: response
+          type: boolean
+          nullable: false
+          since: 2.6
+          doc: 'True if processed successfully, false otherwise'
+  - id: 18
+    name: uploadJobMultipart
+    since: 2.6
+    doc: ''
+    request:
+      retryable: true
+      partitionIdentifier: -1
+      params:
+        - name: sessionId
+          type: UUID
+          nullable: false
+          since: 2.6
+          doc: 'Unique session ID of the job upload request'
+        - name: currentPartNumber
+          type: int
+          nullable: false
+          since: 2.6
+          doc: 'The current part number being sent. Starts from 1'
+        - name: totalPartNumber
+          type: int
+          nullable: false
+          since: 2.6
+          doc: 'The total number of parts to be sent. Minimum value is 1'
+        - name: partData
+          type: byteArray
+          nullable: false
+          since: 2.6
+          doc: 'The binary data of the message part'
+        - name: partSize
+          type: int
+          nullable: false
+          since: 2.6
+          doc: 'The size of binary data'
+        - name: sha256Hex
+          type: String
+          nullable: false
+          since: 2.6
+          doc: 'Hexadecimal SHA256 of the message part'
+    response:
+      params:
+        - name: response
+          type: boolean
+          nullable: false
+          since: 2.6
+          doc: 'True if processed successfully, false otherwise'
```

Comment:
I guess an error about why the upload failed (e.g., hash doesn't match, no space left on storage) will still be sent, won't it?

Reply:
This call returns true it is succeeds, all other errors are propagated with exception.